File: python/clase-14/final/main.py
"""Batch API: crear batch, consultar estado y procesar resultados cuando termine."""

import os
from anthropic import Anthropic
from anthropic.types.message_create_params import MessageCreateParamsNonStreaming
from anthropic.types.messages.batch_create_params import Request

MODEL = "claude-sonnet-4-6"


def main() -> None:
    api_key = os.getenv("ANTHROPIC_API_KEY")
    if not api_key:
        raise RuntimeError("Define ANTHROPIC_API_KEY.")

    client = Anthropic(api_key=api_key)
    batch = client.messages.batches.create(requests=[
        Request(
            custom_id="invoice-001",
            params=MessageCreateParamsNonStreaming(
                model=MODEL,
                max_tokens=500,
                messages=[{"role": "user", "content": "Resume esta factura de ejemplo."}],
            ),
        )
    ])
    print(batch.id, batch.processing_status)

    batch_status = client.messages.batches.retrieve(batch.id)
    print(batch_status.processing_status)

    if batch_status.processing_status == "ended":
        for result in client.messages.batches.results(batch.id):
            print(result.custom_id, result.result.type)
    else:
        print("El batch todavía no termina. Vuelve a consultar más tarde antes de leer results.")


if __name__ == "__main__":
    main()

File: typescript/clase-14/final/main.ts
import Anthropic from "@anthropic-ai/sdk";

export {};

const MODEL = "claude-sonnet-4-6";
const apiKey = process.env.ANTHROPIC_API_KEY;
if (!apiKey) throw new Error("Define ANTHROPIC_API_KEY.");

const client = new Anthropic({ apiKey });
const batch = await client.messages.batches.create({
  requests: [{
    custom_id: "invoice-001",
    params: {
      model: MODEL,
      max_tokens: 500,
      messages: [{ role: "user", content: "Resume esta factura de ejemplo." }],
    },
  }],
});

console.log(batch.id, batch.processing_status);
const batchStatus = await client.messages.batches.retrieve(batch.id);
console.log(batchStatus.processing_status);

if (batchStatus.processing_status === "ended") {
  const results = await client.messages.batches.results(batch.id);
  for await (const result of results) {
    console.log(result.custom_id, result.result.type);
  }
} else {
  console.log("El batch todavía no termina. Vuelve a consultar más tarde antes de leer results.");
}
